Gems/AWSCore/Code/Source/Editor/UI/AWSCoreResourceMappingToolAction.cpp

```diff
@@ -6,133 +6,95 @@
+            !AZ::IO::SystemFile::Exists(m_toolQtBinDirectoryPath.c_str()) ||
+            !AZ::IO::SystemFile::Exists(m_toolConfigDirectoryPath.c_str()) ||
+            !AZ::IO::SystemFile::Exists(m_toolLogDirectoryPath.c_str()))
         {
+            AZ_Error(AWSCoreResourceMappingToolActionName, false,
+                "Expected parameter for tool launch command is invalid, engine python path: %s, tool script path: %s, tool qt binaries path: %s, tool config path: %s, tool log path: %s",
+                m_enginePythonEntryPath.c_str(), m_toolScriptPath.c_str(), m_toolQtBinDirectoryPath.c_str(), m_toolConfigDirectoryPath.c_str(), m_toolLogDirectoryPath.c_str());
             return "";
         }
 
         AZStd::string profileName = "default";
         AWSCoreInternalRequestBus::BroadcastResult(profileName, &AWSCoreInternalRequests::GetProfileName);
 
-        AZStd::string configPath = "";
-        AWSCoreInternalRequestBus::BroadcastResult(configPath, &AWSCoreInternalRequests::GetResourceMappingConfigFolderPath);
-
         if (m_isDebug)
         {
             return AZStd::string::format(
-                "%s debug %s --binaries_path %s --debug --profile %s --config_path %s", m_enginePythonEntryPath.c_str(),
-                m_toolScriptPath.c_str(), m_toolQtBinDirectoryPath.c_str(), profileName.c_str(), configPath.c_str());
+                "\"%s\" debug -B \"%s\" --binaries-path \"%s\" --debug --profile \"%s\" --config-path \"%s\" --log-path \"%s\"",
+                m_enginePythonEntryPath.c_str(), m_toolScriptPath.c_str(), m_toolQtBinDirectoryPath.c_str(),
+                profileName.c_str(), m_toolConfigDirectoryPath.c_str(), m_toolLogDirectoryPath.c_str());
         }
         else
         {
             return AZStd::string::format(
-                "%s %s --binaries_path %s --profile %s --config_path %s", m_enginePythonEntryPath.c_str(),
-                m_toolScriptPath.c_str(), m_toolQtBinDirectoryPath.c_str(), profileName.c_str(), configPath.c_str());
+                "\"%s\" -B \"%s\" --binaries-path \"%s\" --profile \"%s\" --config-path \"%s\" --log-path \"%s\"",
+                m_enginePythonEntryPath.c_str(), m_toolScriptPath.c_str(), m_toolQtBinDirectoryPath.c_str(),
+                profileName.c_str(), m_toolConfigDirectoryPath.c_str(), m_toolLogDirectoryPath.c_str());
         }
     }
 
-    AZStd::string AWSCoreResourceMappingToolAction::GetToolLogPath() const
+    AZStd::string AWSCoreResourceMappingToolAction::GetToolLogFilePath() const
     {
-        return m_toolLogPath;
+        AZ::IO::Path toolLogFilePath = (m_toolLogDirectoryPath / "resource_mapping_tool.log").LexicallyNormal();
+        if (!AZ::IO::SystemFile::Exists(toolLogFilePath.c_str()))
+        {
+            AZ_Error(AWSCoreResourceMappingToolActionName, false, "Invalid tool log file path: %s", toolLogFilePath.c_str());
+            return "";
+        }
+        return toolLogFilePath.c_str();
```

Comment:
Also another small optimization is that the `AZ::IO::Path` contains a `Native` function on it which returns a reference to the `AZStd::string` it stores.
So it is slightly more efficient to return a copy of a string from an `AZStd::string` than a `const char*` because an `AZStd::string` supplies it's string length already.

```suggestion
        return toolLogFilePath.Native();
```